sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#nullable enable
+
+using OpenTelemetry;
+
+namespace Azure.Monitor.OpenTelemetry
+{
+    /// <summary>
+    /// An OpenTelemetry trace processor that tags activities with a session ID
+    /// if a profiling session is active. This enables correlation between
+    /// traces and profiler artifacts.
+    /// </summary>
+    internal sealed class ProfilingSessionTraceProcessor : BaseProcessor<Activity>
+    {
+        /// <summary>
+        /// Name of the Tag (and subsequent OpenTelemetry attribute) that
+        /// records the profiling session ID.
+        /// </summary>
+        internal const string TagName = "ds.profiling.session";
```

Comment:
@rajkumar-rangaraj Need your opinion on the right name for the tag/attribute. Is there a convention?

Reply:
There are no OpenTelemetry semantics for profiling. Think this TagName is good.  Is there a reason to prefix `ds`?

Reply:
@cijothomas Do you have any recommendations here for the tag name?

Reply:
"ds" is for "Diagnostic Services", but that's an internal team name leaking through. How about just "profiling.session" ?

Reply:
`profiling.session` should be good to use.